engine/schema/src/main/java/com/cloud/upgrade/SystemVmTemplateRegistration.java

```diff
@@ -458,10 +466,32 @@ private Long createTemplateObjectInDB(SystemVMTemplateDetails details) {
         template.setState(VirtualMachineTemplate.State.Inactive);
         template.setDeployAsIs(Hypervisor.HypervisorType.VMware.equals(details.getHypervisorType()));
         template = vmTemplateDao.persist(template);
-        if (template == null) {
-            return null;
+        return template;
+    }
+
+    private VMTemplateZoneVO createOrUpdateTemplateZoneEntry(long zoneId, long templateId) {
+        VMTemplateZoneVO templateZoneVO = vmTemplateZoneDao.findByZoneTemplate(zoneId, templateId);
+        if (templateZoneVO == null) {
+            templateZoneVO = new VMTemplateZoneVO(zoneId, templateId, new java.util.Date());
+            templateZoneVO = vmTemplateZoneDao.persist(templateZoneVO);
+        } else {
+            templateZoneVO.setRemoved(GenericDaoBase.DATE_TO_NULL);
```

Comment:
I think, no need to explicitly set removed to null (the record retrieved using _findByZoneTemplate()_ already has removed null)